clang warning (enumeration value 'GDL_UNDEF' not handled in switch)

```
/home/slayoo/devel/gdl/src/basic_fun_jmg.cpp:848:12: warning: enumeration value 'GDL_UNDEF' not handled in switch [-Wswitch]
    switch(type) {
           ^
```

--- a/src/basic_fun_jmg.cpp
+++ b/src/basic_fun_jmg.cpp
@@ -868,6 +868,7 @@ namespace lib {
       else return make_array_template<DULongGDL>(e, dimKey, value, off, inc)->Convert2(GDL_STRING);
     case GDL_STRUCT:
       e->Throw("Invalid type specified for result.");
+    default:; // Default to FLOAT to emulate IDL
     }
 
     return make_array_template<DFloatGDL>(e, dimKey, value, off, inc);

--- a/src/basic_pro.cpp
+++ b/src/basic_pro.cpp
@@ -2192,8 +2192,11 @@ static DWORD launch_cmd(BOOL hide, BOOL nowait,
       AppendIfNeeded(proFile, ".pro");
 
       bool found = CompleteFileName(proFile);
-      if (!found )
-        if (!quiet) e->Throw("Not found: " + proFile); else return;
+      if (!found ) {
+        if (!quiet)
+          e->Throw("Not found: " + proFile);
+        else return;
+      }
 
       // file already opened?
       bool open = false;

--- a/src/hash.cpp
+++ b/src/hash.cpp
@@ -197,6 +197,9 @@ static DStructGDL* GetOBJ( BaseGDL* Objptr, EnvUDT* e)
 			  throw GDLException(  "Object ID <"+i2s(ID)+"> not found.");
 		else ThrowFromInternalUDSub( e, "Object ID <"+i2s(ID)+"> not found.");      
     }
+
+    assert(false);
+    return NULL;
   }
 
 // if not found returns -(pos +1)

--- a/src/list.cpp
+++ b/src/list.cpp
@@ -89,18 +89,19 @@
 #define MAKE_LONGGDL(X, XLong) \
         DLongGDL* XLong=0; \
         Guard<DLongGDL> XLongGuard; \
-        if( X != 0) \
-          if( X->Type() == GDL_LONG) \
+        if(X != 0) { \
+          if(X->Type() == GDL_LONG) \
             XLong = static_cast<DLongGDL*>( X); \
-          else  { \
-            try{ \
+          else { \
+            try { \
               XLong = static_cast<DLongGDL*>( X->Convert2( GDL_LONG, BaseGDL::COPY)); \
-                } \
-                catch( GDLException& ex) { \
-                  ThrowFromInternalUDSub( e, ex.ANTLRException::getMessage()); \
-                } \
+            } \
+            catch( GDLException& ex) { \
+              ThrowFromInternalUDSub( e, ex.ANTLRException::getMessage()); \
+            } \
             XLongGuard.Init( XLong); \
-          }
+          } \
+	}
 
 #include "includefirst.hpp"
 
@@ -176,6 +177,9 @@ namespace lib {
     {
       ThrowFromInternalUDSub( e, "Object ID <"+i2s(ID)+"> not found.");      
     }
+
+    assert(false);
+    return NULL;
   }
  
   DStructGDL* GetLISTStruct( EnvUDT* e, DPtr actP)
@@ -1716,8 +1720,8 @@ BaseGDL* list__toarray( EnvUDT* e)
             {
             if ( theType == GDL_COMPLEXDBL) resultType = GDL_COMPLEXDBL;
             else if (resultType == GDL_COMPLEXDBL) continue;
-            else if (( theType == GDL_COMPLEX) && (resultType == GDL_DOUBLE)
-              || ( theType == GDL_DOUBLE) && (resultType == GDL_COMPLEX) )
+            else if ((theType == GDL_COMPLEX && resultType == GDL_DOUBLE)
+                 || (theType == GDL_DOUBLE && resultType == GDL_COMPLEX))
                 resultType = GDL_COMPLEXDBL;
             else if (resultType == GDL_COMPLEX ) continue;
             else if (theType == GDL_DOUBLE ) resultType = GDL_DOUBLE;

